tests/utils.py

```diff
@@ -258,7 +258,11 @@ def assert_guest_os_memory_amount(vm, spec_memory_amount):
 
 def assert_restart_required_condition(vm, expected_message):
     def _get_restart_required_condition(vm):
-        return [condition for condition in vm.instance.status.conditions if condition.type == "RestartRequired"][0]
+        return [
+            condition
+            for condition in vm.instance.status.conditions
+            if condition.type == "RestartRequired" and condition.status == "True"
+        ][0]
```

Comment:
will fail with `IndexError` now if no `RestartRequired`
it's ok to return empty list here since in line 275 we have `if sample and sample.message == expected_message` (if sample is empty list, second condition is not checked)

Reply:
I just see your comment. I changed it to return None, but empty list makes more sense. I've pushed an update.